openedx/frontend-app-course-authoring
# feat: improve error messaging and empty updates

## Description

The JIRA ticket noted below was the original issue of this PR, but was resolved with work surrounding the data/time converter. However, the logging of the error showed a need for better error messaging and callout when no updates are saved to a course. This change impacts Authors.

Before:
<img width="1422" alt="Screenshot 2024-05-21 at 1 55 45 PM" src="https://github.com/openedx/frontend-app-course-authoring/assets/42981026/4c130f49-5501-4a17-9abc-89e92a1981b3">

After:
<img width="1422" alt="Screenshot 2024-05-21 at 1 54 54 PM" src="https://github.com/openedx/frontend-app-course-authoring/assets/42981026/7be29a7d-16f9-4243-b738-29cbe5f48352">

## Supporting information

JIRA Ticket: [TNL-11572](https://2u-internal.atlassian.net/browse/TNL-11572)
> Reported that authors are unable to edit/change a link in their welcome page. Screenshot of the error message attached below.
![image](https://github.com/openedx/frontend-app-course-authoring/assets/42981026/fe491162-634e-412d-9ebc-94528c4bb4eb)
![image](https://github.com/openedx/frontend-app-course-authoring/assets/42981026/b5a17987-f580-4873-9699-bcb05d3212e6)

## Testing instructions

1. Open a course and navigate to the Course Updates page
2. Confirm that page loads content as expected
3. If the course has updates, delete all.
4. Empty updates should show first time message with add button
5. Add first update
6. For each button click (adding, editing, deleting) force the api to throw an error
7. Confirm that for each action an error alert appears at the top of the page with correct message
